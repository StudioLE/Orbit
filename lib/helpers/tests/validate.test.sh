#!/bin/bash
set -euo pipefail

# ARGS
ACTUAL_FILE=$1

source /srv/lib/helpers/validate.sh
source /srv/lib/helpers/is-int.sh

declare OUTPUT=""

for i in -1 0 1 10 100000 0.5 -0.5 1.0 .0 "" 1
do
  OUTPUT+="$i: "
  set +e
  OUTPUT+=$(validate "is-int" "$i" 2>&1)
  EXIT_CODE=$?
  set -e
  OUTPUT+="; EXIT_CODE: ${EXIT_CODE}"
  OUTPUT+=$'\n'
done

echo "${OUTPUT}" > "${ACTUAL_FILE}"
