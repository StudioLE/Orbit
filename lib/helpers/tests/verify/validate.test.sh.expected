-1: TRUE; EXIT_CODE: 0
0: TRUE; EXIT_CODE: 0
1: TRUE; EXIT_CODE: 0
10: TRUE; EXIT_CODE: 0
100000: TRUE; EXIT_CODE: 0
0.5: ❗  Failed validation is-int: 0.5; EXIT_CODE: 1
-0.5: ❗  Failed validation is-int: -0.5; EXIT_CODE: 1
1.0: ❗  Failed validation is-int: 1.0; EXIT_CODE: 1
.0: ❗  Failed validation is-int: .0; EXIT_CODE: 1
: ❗  Failed validation is-int: ; EXIT_CODE: 1
1: TRUE; EXIT_CODE: 0

